src/main/groovy/edu/oregonstate/mist/trailsapi/resources/TrailsResource.groovy

```diff
@@ -161,14 +168,26 @@ public class TrailsResource extends Resource {
         }
     }
 
+    /**********************************************************************************************
+    Function: zipCodeValidator
+    Description: Checks for validity of a zip code
+    Input: String containing zip code
+    Output: Returns true valid zip code, false otherwise
+    **********************************************************************************************/
+    Boolean zipCodeValidator(String zipCode) {
+        String regex = '^[0-9]{5}(?:-[0-9]{4})?$'
+        Pattern pattern = Pattern.compile(regex)
+        pattern.matcher(zipCode).matches()
```

Comment:
Instead of compiling the pattern each time the method is called, you might be able to declare the regex like this

`String regex = /^[0-9]{5}(?:-[0-9]{4})?$/`

and then check it like

`zipCode ==~ regex`